### PR TITLE
add proxy mode to read non-local volumes

### PR DESCRIPTION
**Pull request by song-zhang**
add support for preview picture in Cross-network environment.

for example:

                  10.128.36.xxx                                                     192.168.1.xxx
user ---------------------------->    nginx         ----------------------------->  weed

redirect ip 192.168.1.xxx is unreachable from 10.128.36.xxx

### CODE REVIEW

File: weed/server/volume_server_handlers_read.go [hunk 1]
```diff
@@ -58,14 +58,53 @@ func (vs *VolumeServer) GetOrHeadHandler(w http.ResponseWriter, r *http.Request)
...
 		}
 		lookupResult, err := operation.Lookup(vs.GetMaster, volumeId.String())
 		glog.V(2).Infoln("volume", volumeId, "found on", lookupResult, "error", err)
-		if err == nil && len(lookupResult.Locations) > 0 {
+		if err != nil || len(lookupResult.Locations) <= 0{
+			glog.V(0).Infoln("lookup error:", err, r.URL.Path)
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+		if vs.ReadMode == "remote" {
```
Comment (chrislusf) [inline weed/server/volume_server_handlers_read.go]: change this mode name to "proxy"?

File: weed/server/volume_server_handlers.go [hunk 2]
```diff
@@ -81,6 +82,7 @@ func getContentLength(r *http.Request) int64 {
 }
 
 func (vs *VolumeServer) publicReadOnlyHandler(w http.ResponseWriter, r *http.Request) {
+	fmt.Printf("publicReadOnlyHandler in.")
```
Comment (chrislusf) [inline weed/server/volume_server_handlers.go]: remove

File: weed/server/volume_server_handlers_read.go [hunk 3]
```diff
@@ -58,14 +58,53 @@ func (vs *VolumeServer) GetOrHeadHandler(w http.ResponseWriter, r *http.Request)
...
+			// proxy client request to target server
```
Comment (chrislusf) [inline weed/server/volume_server_handlers_read.go:74]: the following proxy code seems copied from somwhere. Can it be refactored into a function?

File: weed/command/server.go [hunk 4]
```diff
@@ -107,7 +107,7 @@ func init() {
 	serverOptions.v.indexType = cmdServer.Flag.String("volume.index", "memory", "Choose [memory|leveldb|leveldbMedium|leveldbLarge] mode for memory~performance balance.")
 	serverOptions.v.diskType = cmdServer.Flag.String("volume.disk", "", "[hdd|ssd|<tag>] hard drive or solid state drive or any tag")
 	serverOptions.v.fixJpgOrientation = cmdServer.Flag.Bool("volume.images.fix.orientation", false, "Adjust jpg orientation when uploading.")
-	serverOptions.v.readRedirect = cmdServer.Flag.Bool("volume.read.redirect", true, "Redirect moved or non-local volumes.")
+	serverOptions.v.readMode = cmdServer.Flag.String("volume.readMode", "redirect", "[local|remote|redirect] how to deal with non-local volume: 'not found|read in remote node|redirect volume location'.")
```
Comment (chrislusf) [inline weed/command/server.go:110]: change remote to proxy

File: weed/server/volume_server_handlers_read.go (hunk 1 above)
Comment (song-zhang) [reply weed/server/volume_server_handlers_read.go]: updated

File: weed/server/volume_server_handlers.go (hunk 2 above)
Comment (song-zhang) [reply weed/server/volume_server_handlers.go]: updated

File: weed/server/volume_server_handlers_read.go (hunk 3 above)
Comment (song-zhang) [reply weed/server/volume_server_handlers_read.go:74]: emm... newly coded, not copied.

File: weed/command/server.go (hunk 4 above)
Comment (song-zhang) [reply weed/command/server.go:110]: update

**Pull request merged**